Gems/Prefab/PrefabDependencyViewer/Code/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# If we are on a host platform, we want to add the host tools targets like the PrefabDependencyViewer.Editor target which
+# will also depend on PrefabDependencyViewer.Static
+if(PAL_TRAIT_BUILD_HOST_TOOLS)
+    ly_add_target(
+        NAME PrefabDependencyViewer.Editor.Static STATIC
+        NAMESPACE Gem
+        AUTOMOC
+        FILES_CMAKE
+            prefabdependencyviewer_editor_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Source
+            PUBLIC
+                Include
+        BUILD_DEPENDENCIES
+            PUBLIC
+                AZ::AzToolsFramework
+            PRIVATE
+                Legacy::Editor.Headers
+                Gem::GraphCanvasWidgets
+    )
+
+    ly_add_target(
+        NAME PrefabDependencyViewer.Editor GEM_MODULE
+        NAMESPACE Gem
+        AUTOMOC
+        OUTPUT_NAME Gem.PrefabDependencyViewer.Editor
+        FILES_CMAKE
+            prefabdependencyviewer_editor_shared_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Source
+            PUBLIC
+                Include
+        BUILD_DEPENDENCIES
+            PUBLIC
+                Gem::PrefabDependencyViewer.Editor.Static
+    )
+
+    # By default, we will specify that the above target PrefabDependencyViewer would be used by
+    # Tool and Builder type targets when this gem is enabled.  If you don't want it
+    # active in Tools or Builders by default, delete one of both of the following lines:
+    ly_create_alias(NAME Prefab.Tools    NAMESPACE Gem TARGETS Gem::PrefabDependencyViewer.Editor)
+    ly_create_alias(NAME Prefab.Builders NAMESPACE Gem TARGETS Gem::PrefabDependencyViewer.Editor)
```

Comment:
These aliases should be `PrefabDependencyViewer.Tools` and `PrefabDependencyViewer.Builders`

Reply:
@lawsonamzn recommended to make the aliases more generic because more tools can be added in the future under the same name which would create only 1 dll file and reduce the overhead associated with many dll's

Reply:
There is a problem with this specific alias though and that is it doesn't include the PrefabBuilder.
If anything this alias should be in the `Gems/Prefab/CMakeList.txt` where it would include both the `Gem::PrefabBuilder.Builders` and `Gem::PrefabDependencyViewer.Editor` targets.

First the PrefabBuilder is only used in the AssetProcessor/AssetBuilder so it doesn't need to be part of the .Tools variant
Second the PrefabDependencyViewer only needs to be available in the Editor, so it doesn't need a .Builders variant

### Gems/Prefab/CMakeLists.txt
```
ly_create_alias(NAME Prefab.Tools    NAMESPACE Gem TARGETS Gem::PrefabDependencyViewer.Editor)
ly_create_alias(NAME Prefab.Builders NAMESPACE Gem TARGETS Gem::PrefabBuilder.Builders)
```

In the Gems/Prefab/PrefabDependencyViewer/Code/CMakeLists.txt file, I suggest you modify the alias to this snippet
```suggestion
    ly_create_alias(NAME PrefabDependencyViewer.Tools    NAMESPACE Gem TARGETS Gem::PrefabDependencyViewer.Editor)
```

Reply:
The reason this alias doesn't include the PrefabBuilder.Builders is because PrefabBuilder needs to be always enabled. The reason behind that is that Prefab's are in AzToolsFramework => Prefabs are always enabled => PrefabBuilder needs to always on.

Reply:
The explanation for why the PrefabBuilder doesn't need to be added to the Prefab makes sense, since it is required.

Now for the Builder Alias, you don't actually need the `Gem::PrefabDependencyViewer Editor` target, since the AssetProcessor has no reason to load that MODULE.
You only need the alias in order to get Assets directory added to the AssetProcessor scan directories, therefore the TARGETS option isn't needed
The following is sufficient for that.
```
 ly_create_alias(NAME Prefab.Builders NAMESPACE Gem)
```

Reply:
Resolved